api/src/main/java/ca/bc/gov/educ/api/batchgraduation/controller/JobLauncherController.java

```diff
@@ -54,105 +67,165 @@ public JobLauncherController(JobLauncher jobLauncher, JobRegistry jobRegistry, R
 
     @GetMapping(EducGradBatchGraduationApiConstants.EXECUTE_TVR_RUN_BATCH_JOB)
     @PreAuthorize(PermissionsConstants.RUN_GRAD_ALGORITHM)
-    public void launchTvrRunJob() {
+    @Operation(summary = "Run Manual TVR Job", description = "Run Manual TVR Job", tags = { "TVR" })
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "OK"),@ApiResponse(responseCode = "500", description = "Internal Server Error")})
+    public ResponseEntity<AlgorithmSummaryDTO> launchTvrRunJob() {
         logger.debug("launchTvrRunJob");
         JobParametersBuilder builder = new JobParametersBuilder();
         builder.addLong(TIME, System.currentTimeMillis()).toJobParameters();
-        builder.addString(JOB_TRIGGER, "MANUAL");
-        builder.addString(JOB_TYPE, "TVRRUN");
+        builder.addString(JOB_TRIGGER, MANUAL);
+        builder.addString(JOB_TYPE, TVRRUN);
         try {
-            jobLauncher.run(jobRegistry.getJob("tvrBatchJob"), builder.toJobParameters());
+            JobExecution jobExecution =jobLauncher.run(jobRegistry.getJob("tvrBatchJob"), builder.toJobParameters());
+            ExecutionContext jobContext = jobExecution.getExecutionContext();
+            AlgorithmSummaryDTO summaryDTO = (AlgorithmSummaryDTO)jobContext.get("tvrRunSummaryDTO");
+            return ResponseEntity.ok(summaryDTO);
         } catch (JobExecutionAlreadyRunningException | JobRestartException | JobInstanceAlreadyCompleteException
                 | JobParametersInvalidException | NoSuchJobException e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
+            AlgorithmSummaryDTO summaryDTO = new AlgorithmSummaryDTO();
+            summaryDTO.setException(e.getLocalizedMessage());
+            return ResponseEntity.status(500).body(summaryDTO);
         }
 
     }
 
     @PostMapping(EducGradBatchGraduationApiConstants.LOAD_STUDENT_IDS)
     @PreAuthorize(PermissionsConstants.LOAD_STUDENT_IDS)
-    public void loadStudentIDs(@RequestBody List<LoadStudentData> loadStudentData) {
-        logger.info("Inside loadStudentIDs");
+    @Operation(summary = "Load Students to GRAD", description = "Load Students to GRAD", tags = { "Student" })
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "OK"),@ApiResponse(responseCode = "500", description = "Internal Server Error")})
+    public ResponseEntity<Integer> loadStudentIDs(@RequestBody List<LoadStudentData> loadStudentData) {
+        logger.debug("Inside loadStudentIDs");
         OAuth2AuthenticationDetails auth = (OAuth2AuthenticationDetails) SecurityContextHolder.getContext().getAuthentication().getDetails();
         String accessToken = auth.getTokenValue();
-        restUtils.getStudentByPenFromStudentAPI(loadStudentData, accessToken);
-
+        Integer recordsAdded = restUtils.getStudentByPenFromStudentAPI(loadStudentData, accessToken);
+        if(recordsAdded != null)
+            return ResponseEntity.ok(recordsAdded);
```

Comment:
Hi @sreepadkarekkad, below the recordsAdded will always be null. (line 127). Can you return a reason?